packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -332,15 +336,55 @@ const PipelineWrapper: React.FC<IProps> = ({
   );
 
   const onError = (error?: Error): void => {
-    showDialog({
-      title: 'Load pipeline failed!',
-      body: <p> {error || ''} </p>,
-      buttons: [Dialog.okButton()]
-    }).then(() => {
-      if (shell.currentWidget) {
-        shell.currentWidget.close();
-      }
-    });
+    if (error instanceof PipelineOutOfDateError) {
+      showDialog({
+        title: 'Migrate pipeline?',
+        body: (
+          <p>
+            This pipeline corresponds to an older version of Elyra and needs to
+            be migrated.
+            <br />
+            Although the pipeline can be further edited and/or submitted after
+            its update,
+            <br />
+            the migration will not be completed until the pipeline has been
+            saved within the editor.
+            <br />
+            <br />
+            Proceed with migration?
+          </p>
+        ),
+        buttons: [Dialog.cancelButton(), Dialog.okButton()]
+      }).then(result => {
+        if (result.button.accept) {
+          // proceed with migration
+          console.log('migrating pipeline');
+          setPipeline(
+            migrate(pipeline, pipeline =>
+              PipelineService.setNodePathsRelativeToPipeline(
+                pipeline,
+                contextRef.current.path
+              )
+            )
+          );
+          contextRef.current.model.dirty = true;
```

Comment:
You should avoid calling `setPipeline` directly. We have the react state of the pipeline connected to file changes so using `contextRef.current.model.fromString("<pipeline-string>")` will handle the dirty state and the react pipeline state

Reply:
gotcha, I'll switch to using that